--- src/os/os_test.go.orig	2014-02-20 13:14:45.543644182 -0600
+++ src/os/os_test.go	2014-02-20 13:14:55.934813622 -0600
@@ -854,7 +854,7 @@
 		t.Fatal(err)
 	}
 	defer r.Close()
-	p, err := StartProcess("/bin/hostname", []string{"hostname"}, &ProcAttr{Files: []*File{nil, w, Stderr}})
+	p, err := StartProcess("/usr/bin/hostname", []string{"hostname"}, &ProcAttr{Files: []*File{nil, w, Stderr}})
 	if err != nil {
 		t.Fatal(err)
 	}
