diff -up go-5a45a157f2e94cb3fec38a3be8afa3bffd800067/src/make.bash.verbose go-5a45a157f2e94cb3fec38a3be8afa3bffd800067/src/make.bash
--- go-5a45a157f2e94cb3fec38a3be8afa3bffd800067/src/make.bash.verbose	2017-03-31 09:24:20.361571345 +0200
+++ go-5a45a157f2e94cb3fec38a3be8afa3bffd800067/src/make.bash	2017-03-31 09:29:22.534173555 +0200
@@ -167,7 +167,7 @@ if [ "$GOHOSTARCH" != "$GOARCH" -o "$GOH
 	# CC_FOR_TARGET is recorded as the default compiler for the go tool. When building for the host, however,
 	# use the host compiler, CC, from `cmd/dist/dist env` instead.
 	CC=$CC GOOS=$GOHOSTOS GOARCH=$GOHOSTARCH \
-		"$GOTOOLDIR"/go_bootstrap install -gcflags "$GO_GCFLAGS" -ldflags "$GO_LDFLAGS" -v std cmd
+		"$GOTOOLDIR"/go_bootstrap install -gcflags "$GO_GCFLAGS" -ldflags "$GO_LDFLAGS" -v -x std cmd
 	echo
 fi
 
@@ -175,7 +175,7 @@ echo "##### Building packages and comman
 
 old_bin_files=$(cd $GOROOT/bin && echo *)
 
-CC=$CC_FOR_TARGET "$GOTOOLDIR"/go_bootstrap install $GO_FLAGS -gcflags "$GO_GCFLAGS" -ldflags "$GO_LDFLAGS" -v std cmd
+CC=$CC_FOR_TARGET "$GOTOOLDIR"/go_bootstrap install $GO_FLAGS -gcflags "$GO_GCFLAGS" -ldflags "$GO_LDFLAGS" -v -x std cmd
 
 # Check that there are no new files in $GOROOT/bin other than go and gofmt
 # and $GOOS_$GOARCH (a directory used when cross-compiling).
